modules/ROOT/pages/mule-server-notification.adoc

```diff
@@ -0,0 +1,284 @@
+
+|Mule Context Notification |Can be fired when an event such as the mule context start occurs. |`MuleContextNotificationListener`
+
+|Routing Notification |Async-Reply routers use this when an event is received for an event group that has already been processed. |`RoutingNotificationListener`
+
+|Security Notification |Is fired when a request for authorization failed. |`SecurityNotificationListener`
+
+|Transaction Notification |Is fired after a transaction has begun, was committed, or was rolled back. |`TransactionNotificationListener`
+
+|Async-Message Notification |Can be fired when async work is scheduled and completed for a given flow. |`AsyncMessaheNotificationListener`
+
+|Pipeline-Message Notification |Is fired at key steps in the processing of pipeline. |`PipelineMessageNotificationListener`
+
+|Message-Processor Notification |A message processor was invoked. |`MessageProcessorNotificationListener`
+
+|Exception Strategy Notification |An exception strategy was invoked. |`ExceptionStrategyNotificationListener`
+
+|Extension Notification |Can be fired by extensions, including custom data. |`ExtensionNotificationListener`
+
+|Connector-Message Notification |Is fired when a message is received or sent through a Connector. |`ConnectorMessageNotificationListener`
+
+|===
+
+The listener interfaces all have a single method:
+
+[source,java]
+
+----
+
+public void onNotification(T notification);
+
+----
+
+where T is a notification class (listener class without the 'Listener' at the end).
+
+Depending on the listener implemented, only certain notifications will be received. For example, if the object implements `ManagementNotificationListener`, only notifications of type `ManagementNotification` will be received. Objects can implement more than one listener to receive more types of notifications.
+
+== Registering Listeners Programmatically
+
+You can register listeners as follows:
+
+----
+
+notificationListenerRegistry.registerListener(listener);
+
+----
+
+or:
+
+----
+
+notificationListenerRegistry.registerListener(listener, selector);
+
+----
+
+where `listener` is a `NotificationListener<N>` instance and `selector` is a `Predicate<N>` that works as a filter to apply on a fired notification before calling the listener with it.
+
+== Notification Action Codes
+
+Each notification has an action code that determines the notification type. You can query the action code to determine its type. For example:
+
+*MyObject.java*
+
+[source,java,linenums]
+
+----
+public class MyObject implements ConnectionNotificationListener<ConnectionNotification> {
+
+    (...)
+
+    public void onNotification(ConnectionNotification notification) {
+
+        if (valueOf(ConnectionNotification.CONNECTION_FAILED).equals(notification.getAction().getIdentifier())) {
+
+            System.out.println("Connection failed");
```

Comment:
I would replace the System.out with a comment indicating that the logic for handling the connection failed notification should be placed there.